src/database/sql_database.h

```diff
@@ -270,11 +270,11 @@ class SQLDatabase : public Database {
     /* helper for removeObject(s) */
     void _removeObjects(const std::vector<std::int32_t>& objectIDs);
 
-    std::unique_ptr<ChangedContainers> _recursiveRemove(
+    ChangedContainers _recursiveRemove(
         const std::vector<std::int32_t>& items,
         const std::vector<std::int32_t>& containers, bool all);
 
-    virtual std::unique_ptr<ChangedContainers> _purgeEmptyContainers(const std::unique_ptr<ChangedContainers>& maybeEmpty);
+    virtual std::unique_ptr<ChangedContainers> _purgeEmptyContainers(ChangedContainers maybeEmpty);
```

Comment:
Looks like a copy is created.

Reply:
Not possible. Copy constructor is implicitly deleted because move constructors are defined (yes I tested this). See: https://en.cppreference.com/w/cpp/language/copy_constructor <-- last part under implicitly deleted.